RELEASE.md

```diff
@@ -7,6 +7,13 @@ Please follow the established format:
 - Use present tense (e.g. 'Add new feature')
 - Include the ID number for the related PR (or PRs) in parentheses
 -->
+
+# Release 4.1.1
+## Bug fixes and other changes
+
+- Fix display of boolean values on tracking dataset (#652)
+- Fix node IDs in test data after Kedro updates its _str_logic for node (#653)
```

Comment:
As discussed yesterday, I don't think this needs to go into the release note. It's not userfacing.

Reply:
I see- not sure if we shouldn't include non-user facing notes given that we are open source, moreover us having the practice of including fixes and changes to the server on our previous release notes?